kotlinlib/src/mill/kotlinlib/ksp/KspModule.scala

```diff
@@ -0,0 +1,157 @@
+package mill.kotlinlib.ksp
+
+import mill.*
+import mill.api.{PathRef, Result}
+import mill.define.Task
+import mill.kotlinlib.worker.api.{KotlinWorker, KotlinWorkerTarget}
+import mill.kotlinlib.{Dep, DepSyntax, KotlinModule}
+
+import java.io.File
+
+/**
+ * Sets up the kotlin compiler for using KSP (Kotlin Symbol Processing)
+ * by plugging in the symbol-processing and symbol-processing-api dependencies.
+ *
+ * Use of kotlin-compiler-embedded is also recommended (and thus enabled by default)
+ * to avoid any classpath conflicts between the compiler and user defined plugins!
+ */
+trait KspModule extends KotlinModule {
+
+  def symbolProcessingVersion: T[String]
+
+  /**
+   * Mandatory plugins that are needed for KSP to work.
+   * For more info go to [[https://kotlinlang.org/docs/ksp-command-line.html]]
+   *
+   * @return
+   */
+  def kspPlugins: T[Agg[Dep]] = Task {
+    Agg(
+      ivy"com.google.devtools.ksp:symbol-processing-api:${kotlinVersion()}-${symbolProcessingVersion()}",
+      ivy"com.google.devtools.ksp:symbol-processing:${kotlinVersion()}-${symbolProcessingVersion()}"
+    )
+  }
+
+  def kspProjectBaseDir: T[PathRef] = Task {
+    PathRef(moduleDir)
+  }
+
+  /** KSP output dir */
+  def kspOutputDir: T[PathRef] = Task {
+    PathRef(T.dest / "generated" / "ksp" / "main")
+  }
+
+  /** KSP caches dir */
+  def kspCachesDir: T[PathRef] = Task {
+    PathRef(T.dest / "main")
+  }
```

Comment:
What is this for? This is likely a detail of `generateSourcesWithKSP`.

Reply:
removed